detox/src/client/Client.js

```diff
@@ -29,7 +30,7 @@ class Client {
     this._onUnhandledServerError = this._onUnhandledServerError.bind(this);
     this._logError = this._logError.bind(this);
 
-    this._sessionId = sessionId;
+    this._sessionId = sessionId || uuid.UUID();
```

Comment:
I'm not entirely sure about this but wouldn't it make more sense for the worker (i.e. consumer) to regenerate the UUID per each client instance? Arguably, choosing a random UUID can be considered a client "feature" but I'd feel more comfortable with the `preconfigured-sessionID-or-fallback-to-random-uuid` logic to be contained in one place (i.e. the DetoxWorker).